McCorrections/src/LeptonWeights.cpp

```diff
@@ -399,6 +400,43 @@ double LeptonWeights::GetTriggerEfficiency(EventInfo& eventInfo, bool isData, Di
     return efficiency;
 }
 
+double  LeptonWeights::GetCustomTauSF(const LepCandidate& leg, UncertaintySource unc_source, UncertaintyScale unc_scale)
+{
+    // taken from: https://twiki.cern.ch/twiki/bin/viewauth/CMS/DoubleHiggsToBBTauTauWorkingLegacyRun2#Custom_Tau_ID_SF
+    static const std::map<int, std::map<double, std::map<UncertaintyScale, double>>> tau_sf_error = {
+        { 0, { { 1.078, {{UncertaintyScale::Up, 0.034}, {UncertaintyScale::Down, 0.036} }} } },
+        { 1, { { 1.112, {{UncertaintyScale::Up, 0.023}, {UncertaintyScale::Down, 0.023} }} } },
+        { 10, { { 0.984, {{UncertaintyScale::Up, 0.063}, {UncertaintyScale::Down, 0.067} }} } },
+        { 11, { { 0.759, {{UncertaintyScale::Up, 0.178}, {UncertaintyScale::Down, 0.259} }} } }
+    };
+
+    static const std::map<int, UncertaintySource> dm_unc_sources = {
+         {0, UncertaintySource::TauCustomSF_DM0},
+         {1, UncertaintySource::TauCustomSF_DM1},
+         {10, UncertaintySource::TauCustomSF_DM10},
+         {11, UncertaintySource::TauCustomSF_DM11}
+    };
+
+    if(period == Period::Run2017){
+        if(leg->leg_type() == LegType::tau){
```

Comment:
- custom SFs should be applied only for the true taus
- ifs in L420&421 can be merged into a single if.